test/pkey_meth_kdf_test.c

```diff
@@ -144,6 +144,67 @@ static int test_kdf_hkdf(int index)
     return ret;
 }
 
+static int test_kdf_hkdf_sha256(int index)
+{
+    int ret = 0;
+    EVP_PKEY_CTX *pctx;
+    unsigned char out[10];
+    size_t outlen = sizeof(out);
+
+    if ((pctx = EVP_PKEY_CTX_new_id(NID_HKDF_SHA256, NULL)) == NULL) {
```

Comment:
Do we need to use these algorithms with the EVP_PKEY_CTX API?
You may notice that the number of KDF's that use this API is a limited subset.